hi-ml-azure/src/health_azure/himl.py

```diff
@@ -81,6 +81,49 @@ class AzureRunInfo:
     folder will be uploaded to blob storage regularly during the script run."""
 
 
+def validate_num_nodes(existing_compute_clusters: Dict[str, ComputeTarget], compute_cluster_name: str, num_nodes: int
+                       ) -> None:
+    """
+    Check that the user hasn't requested more nodes than the maximum number of nodes allowed by their compute cluster
+
+    :param existing_compute_clusters: A list of AML ComputeTarget objects in a given AML Workspace
+    :param compute_cluster_name: The name of the specific compute cluster whose upper node limit should be checked
+    :param num_nodes: The number of nodes that the user has requested
+    """
+    compute_cluster = existing_compute_clusters[compute_cluster_name]
```

Comment:
The parameters for this function could be simplified. You are passing in a dictionary and a key, but you use it only to do the lookup. Would be more natural to just pass in the compute cluster.